uoapi/timetable/cli.py

```diff
@@ -122,12 +122,25 @@ def cli(args=None):
         ):
             print(json.dumps(out))
 
-def available(retries):
-    tq = qt.TimetableQuery(retries=retries)
-    with tq as gm:
-        out = {"available": list(tq.available.values())}
-    out["messages"] = gm
-    return out
+def available(retries=2):
+    try:
+        tq = qt.TimetableQuery(retries=retries)
+        with tq as gm:
+            out = {"available": [
+                at
+                for at in map(qt.parse_available, tq.available.keys())
+                if at is not None
+            ]}
+        out["messages"] = gm
```

Comment:
I have no idea what this does

Reply:
I've added comments to help with the relevant bits.  `tq` is a context manager, which connects with the school server on entrance, and disconnects on exit.  It automatically picks up the available terms when it connects, (it needs them anyways) which is accessible in the dict `tq.available` where the keys are the term codes, and the values are the pretty strings.

When entering the context, `tq` will give you a list, where it will append messages (e.g. things like "cannot connect to the school" or "successfully connected to the school") that you may (or may not) care about during and after its operation.  That's what `gm` is.